openupgrade_scripts/scripts/mail/17.0.1.15/pre-migration.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_models_renames = [
+    ("mail.channel", "discuss.channel"),
+    ("mail.channel.member", "discuss.channel.member"),
+    ("mail.channel.rtc.session", "discuss.channel.rtc.session"),
+]
+_tables_renames = [
+    ("mail_channel", "discuss_channel"),
+    ("mail_channel_member", "discuss_channel_member"),
+    ("mail_channel_rtc_session", "discuss_channel_rtc_session"),
+    ("mail_channel_res_groups_rel", "discuss_channel_res_groups_rel"),
+]
+_fields_renames = [
+    (
+        "mail.tracking.value",
+        "mail_tracking_value",
+        "field",
+        "field_id",
+    ),
+]
+_columns_renames = {
+    "discuss_channel_res_groups_rel": [
+        ("mail_channel_id", "discuss_channel_id"),
+    ],
+}
+_columns_copies = {
+    "mail_template": [
+        ("report_template", None, None),
+    ],
+}
+
+
+def _discuss_channel_fill_allow_public_upload(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE discuss_channel
+        ADD COLUMN IF NOT EXISTS allow_public_upload BOOLEAN;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE discuss_channel
+        SET allow_public_upload = True
+        WHERE channel_type = 'livechat'
+        """,
+    )
+
+
+def _mail_alias_fill_multiple_values(env):
+    """
+    We will fill value for alias_full_name in post because alias_domain has not been
+    present yet
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mail_alias
+        ADD COLUMN IF NOT EXISTS alias_full_name VARCHAR,
+        ADD COLUMN IF NOT EXISTS alias_incoming_local BOOLEAN,
+        ADD COLUMN IF NOT EXISTS alias_status VARCHAR;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_alias
+        SET
+        alias_incoming_local = True,
+        alias_status = 'valid'
+        """,
+    )
+
+
+def _mail_tracking_value_update_monetary_tracking_values(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_tracking_value
+            SET old_value_float = old_value_monetary,
+                new_value_float = new_value_monetary
+        WHERE old_value_monetary IS NOT NULL
+            OR new_value_monetary IS NOT NULL;
+        """,
+    )
+
+
+def _mail_gateway_allowed(env):
+    """Set some dummy value so that the not null constraint can be created"""
+    env.cr.execute(
+        """
+        UPDATE mail_gateway_allowed SET email='admin@example.com'
+        WHERE email IS NULL"
+        """
```

Comment:
@remi-filament I think we should fill this with a dummy value after all, otherwise Odoo won't be able to set up the not null constraint